### DISCUSSION TITLE
Get Client API system/info data via channel (javascript)

### DISCUSSION THREAD

**Discussion by Rococo69** [nextgenhealthcare/connect#6101] [General]
Hi,

As I have posted [in the forum](https://forums.mirthproject.io/forum/mirth-connect/support/184631-get-client-api-system-info-data-via-channel-javascript#post184631) , I have a channel using the client API to get some information about the server. However, I should be able to access this information via the channel without using the API. Is there any donkey class or any other class that would give me the following :

```
<com.mirth.connect.model.SystemInfo>
<jvmVersion>1.8.0_172</jvmVersion>
<osName>Mac OS X</osName>
<osVersion>10.14.5</osVersion>
<osArchitecture>x86_64</osArchitecture>
<dbName>PostgreSQL</dbName>
<dbVersion>9.6.15</dbVersion>
</com.mirth.connect.model.SystemInfo>
```

I tried to have a look on the source code to see what the API is doing but to be honest I was a bit lost.

**Comment by pacmano1** [comment]
if you post here and the forums, please reference that fact.

**Comment by Rococo69** [comment]
After a long journey through the code, I started searching for system/info references. This took me  to com/mirth/connect/model)
/SystemInfo.java
After that I ended up at com/mirth/connect/server/api/servlets/SystemServlet.java
And finally got my answer. 
```
SystemInfo info = new SystemInfo();
                info.setDbName(metaData.getDatabaseProductName());
                info.setDbVersion(metaData.getDatabaseProductVersion());
                info.setJvmVersion(System.getProperty("java.version"));
                info.setOsName(System.getProperty("os.name"));
                info.setOsVersion(System.getProperty("os.version"));
                info.setOsArchitecture(System.getProperty("os.arch"));
                return info;
```

Inside the channel with javascript:

`java.lang.System.getProperty("java.version")`


Next step is to figure out the metaData information.
